src/utils/check.ts

```diff
@@ -0,0 +1,63 @@
+import isEmpty from 'lodash/isEmpty';
+import isPlainObject from 'lodash/isPlainObject';
+import type { DataType } from '..';
+import { getOperators } from './format';
+// eslint-disable-next-line import/order -- caused by temporarily mixing require with import
+import { Where } from './sequelize-method';
+
+const DataTypes = require('../data-types');
+
+export function isPrimitive(val: any): val is string | number | boolean {
+  const type = typeof val;
+
+  return ['string', 'number', 'boolean'].includes(type);
+}
+
+export function isColString(value: string): boolean {
+  return (
+    typeof value === 'string'
+    && value.startsWith('$')
+    && value.endsWith('$')
+  );
+}
+
+export function canTreatArrayAsAnd(arr: unknown[]): arr is Array<object | Where> {
+  return arr.some(arg => isPlainObject(arg) || arg instanceof Where);
+}
+
+/**
+ * Determine if the default value provided exists and can be described
+ * in a db schema using the DEFAULT directive.
+ *
+ * @param value Any default value.
+ * @returns yes / no.
+ * @private
+ */
+export function defaultValueSchemable(value: DataType): boolean {
+  if (value === undefined) {
+    return false;
+  }
+
+  // TODO this will be schemable when all supported db
+  //  have been normalized for this case
+  if (value instanceof DataTypes.NOW) {
+    return false;
+  }
+
+  if (value instanceof DataTypes.UUIDV1 || value instanceof DataTypes.UUIDV4) {
+    return false;
+  }
+
+  return typeof value !== 'function';
+}
+
+/**
+ * Returns true if a where clause is empty, even with Symbols
+ *
+ * @param obj
+ * @returns
```

Comment:
Boolean too

Reply:
We're not specifying types in the jsdoc anymore since TypeDocs will extract the typing information from typescript, but I think we could go with

```
@returns true if a where clause is empty, even with Symbols ; false otherwise
```

Reply:
I don't really mind either way tbh. You choose

Reply:
Based on what I see, I think it's better to remove the `@returns` altogether if it doesn't add anything that's not already in the description

![image](https://user-images.githubusercontent.com/1280915/153564697-60020396-5958-43f2-8c69-f959049c22e0.png)